libs/jwst-codec/src/doc/codec/content.rs

```diff
@@ -170,7 +170,7 @@ impl Content {
         match self {
             Self::Deleted(len) => *len,
             Self::JSON(strings) => strings.len() as u64,
-            Self::String(string) => string.len() as u64,
+            Self::String(string) => string.chars().count() as u64,
```

Comment:
I'm not sure, does yjs receive a utf-16 length?

Reply:
https://github.com/yjs/yjs/blob/5db1eed181b70cb6a6d7eab66c7e6d752f70141a/src/structs/ContentString.js#L29-L31

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split

yjs use native string.split function, it use utf-16 split the text
this behavior is same as chars().len()

Reply:
Maybe try:
```JavaScript
console.log('🤩'.length) // 2
```
compare with
```rust
println!("{}", "🤩".chars().count()); // 1
```

Reply:
> > yjs use native string.length
> 
> get that, will the following example cause some problem?
> 
> ```js
> console.log('🤩'.length) // 2
> ```
> 
> compared with
> 
> ```rust
> println!("{}", "🤩".chars().count()); // 1
> ```
> 
> How about considering
> 
> ```rust
> string.chars()
>         .into_iter()
>         .fold(0, |len, ch| len + ch.len_utf16());
> ```

lgtm

Reply:
actually we should think about few things:

1. This is a rust backed library and rust does not allow string indexing behavior `str[index]`, because each char is compounded with different length of bytes, so the chars can only be known in iteration.
2. JS will be only one of consumer which will provider service for
3. even in JS, length of 🤩 is 2, but text insert in between will have unexpected result
```js
a = '🤩' // a.length == 2
b = a.split('') // [ '\ud83e', '\udd29' ]
b.join('') // '🤩'
b.splice(1, 0, ' ') // text.insert(1, ' ')
b.join('') // '\ud83e \udd29'
```

That's why I named the position argument in `text::insert()` `char_index` instead of `index`, it makes more sense